solutions/src/main/java/ru/sberbank/school/task08/KryoManager.java

```diff
@@ -0,0 +1,76 @@
+package ru.sberbank.school.task08;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import lombok.NonNull;
+import ru.sberbank.school.task08.state.GameObject;
+import ru.sberbank.school.task08.state.InstantiatableEntity;
+import ru.sberbank.school.task08.state.MapState;
+import ru.sberbank.school.util.Solution;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Solution(8)
+public class KryoManager extends SaveGameManager<MapState<GameObject>, GameObject> {
+
+    private Kryo kryo;
+
+    public KryoManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+        kryo = new Kryo();
+
+        kryo.register(MapState.class, new KryoSerializer());
+        kryo.register(ArrayList.class);
+        kryo.register(GameObject.class);
+        kryo.register(InstantiatableEntity.Status.class);
+        kryo.register(InstantiatableEntity.Type.class);
+    }
+
+    @Override
+    public void saveGame(@NonNull String filename, @NonNull MapState<GameObject> gameState)
+            throws SaveGameException {
+
+            try (FileOutputStream fos = new FileOutputStream(filesDirectory + File.separator + filename);
+                 Output out = new Output(fos)) {
+                kryo.writeObject(out, gameState);
+            } catch (FileNotFoundException ex) {
+                throw new SaveGameException("Файл не найден", ex, SaveGameException.Type.USER, gameState);
+            } catch (IOException ex) {
+                throw new SaveGameException("Ошибка записи", ex, SaveGameException.Type.IO, gameState);
```

Comment:
&#39;catch&#39; child has incorrect indentation level 16, expected level should be 12.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-745274739>*